src/commands/data/schemas/sync.ts

```diff
@@ -10,16 +10,49 @@ export const command = 'sync';
 export const desc = 'Sync all schemas in a directory with the ExH cloud';
 export const builder = (yargs: any) => epilogue(yargs).options({
   dir: {
-    demandOption: true,
+    demandOption: false,
     describe: 'Directory containing the schemas which need to be synced',
     type: 'string',
   },
+  file: {
+    demandOption: false,
+    describe: 'File containing a schema which needs to be synced',
+    type: 'string',
+  },
 });
 
-export const handler = async ({ sdk, dir }) => {
-  await syncTargetDir(sdk, path.resolve(dir || '.'));
+export const handler = async ({ sdk, dir, file }) => {
+  if (!file && !dir) {
```

Comment:
Food for thought, but it is correct that we have added a new --file argument rather than creating a new path argument or something that resolves internally to being a dir or file ? Now as user can hypothetically provide both arguments, but only the dir will sync :thinking_face:

Reply:
Could be that we deprecate dir, leave it in the code and then remove it with the next major release?

Reply:
Allignment through call:
- Merge as is
- Further improvement would be to deprecate `--file` and `--dir` params and use `--path` param instead to cover both cases. **TBD**